pkg/client/collector.go

```diff
@@ -1,50 +1,36 @@
 package loki
 
 import (
-	"fmt"
-	"io"
-	"log"
-	"net/http"
 	"sync"
 
-	"github.com/apache/thrift/lib/go/thrift"
-	"github.com/openzipkin/zipkin-go-opentracing/_thrift/gen-go/zipkincore"
+	"github.com/weaveworks-experiments/loki/pkg/model"
 )
 
 // Want to be able to support a service doing 100 QPS with a 15s scrape interval
 var globalCollector = NewCollector(15 * 100)
 
 type Collector struct {
 	mtx      sync.Mutex
-	traceIDs map[int64]int // map from trace ID to index in traces
-	traces   []trace
+	traceIDs map[uint64]int // map from trace ID to index in traces
+	traces   []model.Trace
 	next     int
 	length   int
 }
 
-type trace struct {
-	traceID int64
-	spans   []*zipkincore.Span
-}
-
 func NewCollector(capacity int) *Collector {
 	return &Collector{
-		traceIDs: make(map[int64]int, capacity),
-		traces:   make([]trace, capacity, capacity),
+		traceIDs: make(map[uint64]int, capacity),
```

Comment:
fyi zipkin trace ids since a few months ago can be 128bit. in thrift format, the representation is high/low. in json it  is simply a longer hex